hack/generator/pkg/codegen/pipeline/testdata/TestInjectJsonSerializationTests/microsoft.person-v20200101_test.golden

```diff
@@ -0,0 +1,198 @@
+	}
+
+	// Check for outcome
+	match := cmp.Equal(subject, actual, cmpopts.EquateEmpty())
+	if !match {
+		actualFmt := pretty.Sprint(actual)
+		subjectFmt := pretty.Sprint(subject)
+		result := diff.Diff(subjectFmt, actualFmt)
+		return result
+	}
+
+	return ""
+}
+
+//Generator of Person instances for property testing - lazily instantiated by
+//PersonGenerator()
+var personGenerator gopter.Gen
+
+// PersonGenerator returns a generator of Person instances for property testing.
+func PersonGenerator() gopter.Gen {
+	if personGenerator != nil {
+		return personGenerator
+	}
+
+	generators := make(map[string]gopter.Gen)
+	AddRelatedPropertyGeneratorsForPerson(generators)
+	personGenerator = gen.Struct(reflect.TypeOf(Person{}), generators)
+
+	return personGenerator
+}
+
+// AddRelatedPropertyGeneratorsForPerson is a factory method for creating gopter generators
+func AddRelatedPropertyGeneratorsForPerson(gens map[string]gopter.Gen) {
+	gens["Spec"] = PersonSpecGenerator()
+	gens["Status"] = PersonStatusGenerator()
+}
+
+func Test_Person_Spec_WhenSerializedToJson_DeserializesAsEqual(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+	parameters.MaxSize = 10
+	properties := gopter.NewProperties(parameters)
+	properties.Property(
+		"Round trip of Person_Spec via JSON returns original",
+		prop.ForAll(RunJSONSerializationTestForPersonSpec, PersonSpecGenerator()))
+	properties.TestingRun(t, gopter.NewFormatedReporter(true, 240, os.Stdout))
+}
+
+// RunJSONSerializationTestForPersonSpec runs a test to see if a specific instance of Person_Spec round trips to JSON and back losslessly
+func RunJSONSerializationTestForPersonSpec(subject Person_Spec) string {
+	// Serialize to JSON
+	bin, err := json.Marshal(subject)
+	if err != nil {
+		return err.Error()
+	}
+
+	// Deserialize back into memory
+	var actual Person_Spec
+	err = json.Unmarshal(bin, &actual)
+	if err != nil {
+		return err.Error()
+	}
+
+	// Check for outcome
+	match := cmp.Equal(subject, actual, cmpopts.EquateEmpty())
+	if !match {
+		actualFmt := pretty.Sprint(actual)
+		subjectFmt := pretty.Sprint(subject)
+		result := diff.Diff(subjectFmt, actualFmt)
+		return result
+	}
+
+	return ""
+}
+
+//Generator of Person_Spec instances for property testing - lazily instantiated by
+//PersonSpecGenerator()
+var personSpecGenerator gopter.Gen
+
+// PersonSpecGenerator returns a generator of Person_Spec instances for property testing.
+func PersonSpecGenerator() gopter.Gen {
+	if personSpecGenerator != nil {
+		return personSpecGenerator
+	}
+
+	generators := make(map[string]gopter.Gen)
+	AddIndependentPropertyGeneratorsForPersonSpec(generators)
+	personSpecGenerator = gen.Struct(reflect.TypeOf(Person_Spec{}), generators)
+
+	return personSpecGenerator
+}
+
+// AddIndependentPropertyGeneratorsForPersonSpec is a factory method for creating gopter generators
+func AddIndependentPropertyGeneratorsForPersonSpec(gens map[string]gopter.Gen) {
+	gens["FamilyName"] = gen.AlphaString()
```

Comment:
Is there not a way to configure a default set of generators, so that we don't need to call out each and every property?

I could imagine these files getting quite large for some of the larger types. I guess we already have a lot of large files so what's one more? Still, it feels like there should be a gen.ReflectGenerator that has some basic data-type limited generators (strings can be any valid string, ints can be valid int, floats any valid float, etc). Then you could ideally just override the ones that needed special handling... which in the case of these tests would maybe be enum types? Maybe not even those.

Reply:
Not as far as I know. If you don't explicitly create a gopter generator for a field, it's ignored by design.

Reply:
I think [gopter arbitrary](https://pkg.go.dev/github.com/leanovate/gopter/arbitrary) is this?

> Package arbitrary contains helpers to create contexts of arbitrary values, i.e. automatically combine generators as needed using reflection.

There are some examples of usage [here](https://github.com/leanovate/gopter/blob/master/arbitrary/example_arbitrary_struct_test.go)? It feels like it might save a _lot_ of code?

Reply:
I'll have a look and might be able to followup with a simplifying PR.